lib/surro-gate/pair.rb

```diff
@@ -7,11 +7,31 @@ class Pair
     def initialize(left, right)
       @left = left
       @right = right
-      @rd_rdy = @wr_rdy = false
+      unmark
     end
 
     def ready?
       @rd_rdy && @wr_rdy || @left.closed? || @right.closed?
     end
+
+    def marked_rd?
```

Comment:
I commented before...

you have this concept of I just wrote, now can I read? and I just read, now can I write?

Think it would be nice to have that as a method.
(then again, if you type that up, it may look like a horrible idea)